fix:(@desktop/keystone): Request saving password to keystone on a startup after changing the password through settings

mprakhov opened the pull request:
### What does the PR do

1. If the user has biometrics enabled and changes the password, we need to let the user enter a current password, check it and if it's valid we need to store it to the keychain and update the `storeToKeychain` prop to `store`
2. Button rename

Fixes: #7349

### Affected areas

- changing the password
- login via biometrics

### Screenshot of functionality (including design for comparison)

https://user-images.githubusercontent.com/117639195/208120591-91918053-5252-4982-a311-91582bbb8485.mov

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 915acd49 | [#1](https://ci.status.im/job/status-desktop/job/branches/job/imports/job/PR-8836/1/) | 2022-12-16 14:50:49 | ~4 min | `imports` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/branches/job/imports/job/PR-8836/1//consoleText) |

saledjenic commented on ui/app/AppLayouts/Profile/popups/ChangePasswordModal.qml:
```diff
@@ -24,9 +24,8 @@ StatusModal {
-                localAccountSettings.storeToKeychainValue = Constants.keychain.storedValue.store;
-                root.privacyStore.storeToKeyChain(d.passwordProcessing);
+            if (Qt.platform.os === "osx" && localAccountSettings.storeToKeychainValue !== Constants.keychain.storedValue.never) {
+                localAccountSettings.storeToKeychainValue = Constants.keychain.storedValue.notNow;
```
@mprakhov maybe we could remove password from the keychain here?

mprakhov replied:
does it really make sense, if we will force the user to restart the application and update the pass in the keychain during the next application start?

saledjenic commented on ui/app/AppLayouts/Profile/popups/ChangePasswordModal.qml:
```diff
@@ -86,7 +85,7 @@ StatusModal {
             id: submitBtn
             objectName: "changePasswordModalSubmitButton"
-            text: qsTr("Change Password")
+            text: qsTr("Change password and restart Status")
```
Don't we need to use `&` instead of `and` here?

mprakhov replied:
Accordingly to the design - no

saledjenic commented:
Cannot remember 100% now, but it could be that we want to enable biometrics on password change and offer user to store his new password to keychain always when he changes it (even if the biometrics option is disabled in the moment of changing password). @anastasiyaig do you remember this?

mprakhov commented:
> @mprakhov we don't need to bother user with store to keychain popup during onboarding, but postpone that for a while for the moment when he logs in, that's why it was part of `mainModule` and you moved it now to `startupModule`.

Store to keychain will be shown only when the user will be logged in (`mainDidLoad` will execute only when we logged in).
IMHO, when I read the code, `startupModule` does not tell me, that this is an onboarding module (onboarding for me - it is when the user does not log in to the application but creates a new account and etc), it tells me, that this is part of the code, which must be executed during the application start. 
Enter the password - inside the startup module, not inside `mainModule`. So when the password is correct, we must save it to the keychain inside `startupModule`. Also if you'll decide to do it inside `mainModule`, you'll need to extract the password from `startupModule` controller (or maybe there are any other ways to do it)

mprakhov commented:
> Cannot remember 100% now, but it could be that we want to enable biometrics on password change and offer user to store his new password to keychain always when he changes it (even if the biometrics option is disabled in the moment of changing password). @anastasiyaig do you remember this?

From the ticket:
`2.2. if user has biometrics enabled, we don't update the old password with the new one, but let user see the prompt that forces him to logout and offer him on the next app run to update the keychain password with the new one`

Also you wrote:
`if password change went well (menas the db is re-encrypted using new pass) we just need to set storeToKeychain prop locally to notNow and display a popup to a user where the only option will be Sign out & Quit and that's the only thing user may do from that state. That further means on next app run, when user logs in using new password we offer him to store that new password to a keychain (that's something which is already in place).`

saledjenic commented:
@mprakhov exactly as I wrote [here](https://github.com/status-im/status-desktop/issues/7349#issuecomment-1353576668), but what I was wondering and ask in my previous comment is that I am not sure about requirement do we need to enable biometrics after password has been changed for a user who has not had biometrics enabled with the aim to ask him if he wants to store a new password in the Keychain on the next app run.

Looking at your recording I see that store to Keychain is displayed while loading app animation was still in progress, that's why I said we should offer store to Keychain once user lends in the app, but in general I don't have anything against moving a logic for displaying keychain popup to startup module unless we are sure that app requirements do not require to store pass/pin to the Keychain from some other parts of the app.

mprakhov commented:
@saledjenic For me, you was the only person who described the expected behavior 😄 , so could you please tell me, to whom can I clarify the expected behavior for the case when the user reset the password with disabled biometrics and restart the application

saledjenic commented:
@mprakhov yes, that's true and when I was working on that was collecting requirements around as well :)
I described those requirements I am sure about, not sure if something has been changed in meantime, but anyhow that's not a big deal, by my opinion both approaches are good. That's why I asked @anastasiyaig if she can remember?

mprakhov commented:
ok, meanwhile I removed unused signal and fixed showing Keychain only when the main window is launched
If there are no other objections, please approve

anastasiyaig commented:
> Cannot remember 100% now, but it could be that we want to enable biometrics on password change and offer user to store his new password to keychain always when he changes it (even if the biometrics option is disabled in the moment of changing password). @anastasiyaig do you remember this?

i dont remember to be honest and since we dont have requirements written down, it is hard to deny / confirm
